source/adios2/core/ADIOS.cpp

```diff
@@ -0,0 +1,88 @@
+/*
+ * Distributed under the OSI-approved Apache License, Version 2.0.  See
+ * accompanying file Copyright.txt for details.
+ *
+ * ADIOS.cpp
+ *
+ *  Created on: Sep 29, 2016
+ *      Author: William F Godoy godoywf@ornl.gov
+ */
+
+#include "ADIOS.h"
+
+/// \cond EXCLUDE_FROM_DOXYGEN
+#include <fstream>
+#include <ios> //std::ios_base::failure
+#include <iostream>
+#include <sstream>
+#include <utility>
+/// \endcond
+
+namespace adios
+{
+
+ADIOS::ADIOS(const bool debugMode) : m_DebugMode(debugMode) {}
+
+ADIOS::ADIOS(const std::string configFile, const bool debugMode)
+: m_ConfigFile(configFile), m_DebugMode(debugMode)
+{
+}
+
+ADIOS::ADIOS(const std::string configFile, MPI_Comm mpiComm,
+             const bool debugMode)
+: m_MPIComm(mpiComm), m_ConfigFile(configFile), m_DebugMode(debugMode)
+{
+    if (m_DebugMode == true)
```

Comment:
Typically you want to use a bool as-is and avoid the extra test.  It also works better when the coding guidelines enforce "boolean variable names should be verbs" rules, i.e. `m_DebugMode` becomes `m_IsDebugMode` or something similar; the boolean methods already do this and it makes reading the if statement very natural: "if obj is ready then do thing" becomes `if (obj->IsReady()) { DoThing(); }`.  It may be something to consider implementing in the near future, I wouldn't push for it at this point though.  Maybe wait till things settle down a bit.